Migrationstool (#5076)
public-ui/kolibri: packages/tools/kolibri-cli/README.md

# KoliBri - CLI-Tools

## Motivation

The `KoliBri` CLI-Tools are a collection of tools to support the development with `KoliBri` components.

## Installation

You can install the `KoliBri` CLI-Tools with `npm`, `pnpm` or `yarn`.

```bash
npm i -g @public-ui/kolibri-cli
pnpm i -g @public-ui/kolibri-cli
yarn add -g @public-ui/kolibri-cli
```

## Usage

Use the `kolibri` command to start the CLI.

```bash
kolibri --help
```

### Migrate

With the `migrate` command you can migrate your project to the latest version of `KoliBri`.

> Actually the following migrations are available:

Reviewer: ```suggestion
Currently the following migrations are available:
```

- Component renaming ✓
- Component removal ⏰
- Property renaming ✓
- Property removal ✓
- Property type change ⏰
- Logic refactoring ⏰
- Expert-Slot refactoring ⏰
- `.vscode/settings.json` add IntelliSense for HTML ⏰
- `.gitignore` exclude `.kolibri.migrate.json` ⏰
- `.tsconfig` add `@public-ui/components` to `types` array ⏰

#### How does it work?

> 1. The migration command will check your project for clear `git history` and the `installed version` of `KoliBri`. Now it load all available migration tasks.

Reviewer: ```suggestion
1. The migration command will check your project for clear `git history` and the `installed version` of `KoliBri`. Now it loads all available migration tasks.
```

> 2. If the task in the correct version range, it will be executed them one by one. Otherwise it will be skipped.

Reviewer: ```suggestion
2. Tasks in the correct version range will be executed one by one. Otherwise they will be skipped.
```

3. After that the `package.json` will be updated with the new version of `KoliBri` and execute the `npm install` command.

> 4. Are there any pending tasks, the migration command will be executed again. Otherwise the migration is finished.

Reviewer: ```suggestion
4. If there are any pending tasks, the migration command will be executed again. Otherwise the migration is finished.
```

			"kol-input-text-rename-property-_list-to-_suggestions": true,
			"kol-link-group-rename-property-_aria-label-to-_label": true,
			"kol-link-group-remove-property-_ordered": true,
			"kol-nav-rename-property-_aria-label-to-_label": true,
			"kol-nav-rename-property-_compact-to-_hide-label": true,
			"kol-nav-remove-property-_has-compact-button": true,
			"kol-pagination-rename-property-_count-to-_total": true,
			"kol-progress-rename-property-_type-to-_variant": true,
			"kol-quote-rename-property-_caption-to-_label": true,
			"kol-select-rename-property-_height-to-_rows": false,
			"kol-select-rename-property-_list-to-_options": false,
			"kol-skip-nav-rename-property-_aria-label-to-_label": true,
			"kol-span-rename-property-_icon-only-to-_hide-label": true,
			"kol-split-button-remove-property-_aria-label": true,
			"kol-table-rename-property-_caption-to-_label": true,
			"kol-tabs-rename-property-_aria-label-to-_label": true,
			"kol-tabs-rename-property-_tab-align-to-_align": true,
			"kol-toast-rename-property-_heading-to-_label": true,
			"kol-version-rename-property-_version-to-_label": true
		}
	}
}
```

## Troubleshooting

> If the migration failed, you can reset the migration with `git reset --hard HEAD~1`.

Reviewer 2: Ich würde es noch erweitern – man muss nicht alles wegschmeißen, nur weil eine Datei falsch migriert wurde:

"If the migration failed, you can reset the migration with `git reset --hard HEAD~1` or by discarding the affected files."

Use the configuration (`.kolibri.config.json`) to exclude some tasks.

Maybe it can help to prepare your code in the tricky places for migration.

Please give us feedback, if you have problems with the migration: [GitHub Issues](https://github.com/public-ui/kolibri/issues/new?assignees=&labels=useful+hint&projects=&template=7_feedback.md&title=%F0%9F%92%A1+CLI%3A+)

## Dry run

You have always the possibility of a dry run. Because before the migration will be executed, you need a clean git history of you project.

After the migration you can check the result with `git status` and `git diff`.

> Is anything wrong, you can reset the migration with `git reset --hard HEAD~1`.

Reviewer 2: Hier würde ich es ebenfalls erweitern:

"Is anything wrong, you can reset the migration with `git reset --hard HEAD~1` or by discarding the affected files."

Reviewer: Ich hätte noch eine Anmerkung zur Grammatik:
`If anything is wrong, you can reset the migration [...]`
